# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/clifford/xmp/gcc34.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./src/loaders/it_load.c.orig	2000-12-29 19:08:28.000000000 +0100
+++ ./src/loaders/it_load.c	2004-03-11 12:38:20.459060960 +0100
@@ -327,14 +327,14 @@
 
 #define BUILD_ENV(X) { \
 	    fread (&env, 1, sizeof (env), f); \
-	    xxih[i].##X##ei.flg = env.flg & IT_ENV_ON ? XXM_ENV_ON : 0; \
-	    xxih[i].##X##ei.flg |= env.flg & IT_ENV_LOOP ? XXM_ENV_LOOP : 0; \
-	    xxih[i].##X##ei.flg |= env.flg & IT_ENV_SLOOP ? XXM_ENV_SUS : 0; \
-	    xxih[i].##X##ei.npt = env.num; \
-	    xxih[i].##X##ei.sus = env.slb; \
-	    xxih[i].##X##ei.sue = env.sle; \
-	    xxih[i].##X##ei.lps = env.lpb; \
-	    xxih[i].##X##ei.lpe = env.lpe; \
+	    xxih[i].X##ei.flg = env.flg & IT_ENV_ON ? XXM_ENV_ON : 0; \
+	    xxih[i].X##ei.flg |= env.flg & IT_ENV_LOOP ? XXM_ENV_LOOP : 0; \
+	    xxih[i].X##ei.flg |= env.flg & IT_ENV_SLOOP ? XXM_ENV_SUS : 0; \
+	    xxih[i].X##ei.npt = env.num; \
+	    xxih[i].X##ei.sus = env.slb; \
+	    xxih[i].X##ei.sue = env.sle; \
+	    xxih[i].X##ei.lps = env.lpb; \
+	    xxih[i].X##ei.lpe = env.lpe; \
 	    if (env.num) xx##X##e[i] = calloc (4, env.num); \
 	    for (j = 0; j < env.num; j++) { \
 		L_ENDIAN16 (env.node[j].x); \
--- ./src/xmms/plugin.c.orig	2001-02-04 01:47:38.000000000 +0100
+++ ./src/xmms/plugin.c	2004-03-11 12:48:04.595258760 +0100
@@ -284,7 +284,7 @@
 	xmp_cfg.filter = TRUE;
 	xmp_cfg.pan_amplitude = 80;
 
-#define CFGREADINT(x) xmms_cfg_read_int (cfg, "XMP", #x, &xmp_cfg.##x)
+#define CFGREADINT(x) xmms_cfg_read_int (cfg, "XMP", #x, &xmp_cfg.x)
 
 	filename = g_strconcat(g_get_home_dir(), "/.xmms/config", NULL);
 	if ((cfg = xmms_cfg_open_file(filename))) {
@@ -668,12 +668,12 @@
 
 	/* Options */
 
-#define OPTCHECK(w,l,o) {						\
+#define OPTCHECK(w,l,ox) {						\
 	w = gtk_check_button_new_with_label(l);				\
 	gtk_object_set_data(GTK_OBJECT(xmp_conf_window), #w, w);	\
 	gtk_widget_show(w);						\
 	gtk_box_pack_start(GTK_BOX(vbox6), w, TRUE, TRUE, 0);		\
-	if (xmp_cfg.##o == 1)						\
+	if (xmp_cfg.ox == 1)						\
 		gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(w), TRUE); \
 }
 
@@ -773,7 +773,7 @@
 	if (!cfg)
 		cfg = xmms_cfg_new();
 
-#define CFGWRITEINT(x) xmms_cfg_write_int (cfg, "XMP", #x, xmp_cfg.##x)
+#define CFGWRITEINT(x) xmms_cfg_write_int (cfg, "XMP", #x, xmp_cfg.x)
 
 	CFGWRITEINT (mixing_freq);
 	CFGWRITEINT (force8bit);
